packages/hooks/src/useDtraderV2Flag.ts

```diff
@@ -1,28 +1,32 @@
-import { useState, useEffect } from 'react';
-import useIsGrowthbookIsLoaded from './useIsGrowthbookLoaded';
-import { isDTraderV2, routes } from '@deriv/shared';
+import { useEffect, useState } from 'react';
+
+import { isDtraderV2DesktopEnabled, isDtraderV2MobileEnabled } from '@deriv/shared';
 import { useDevice } from '@deriv-com/ui';
-import { Analytics } from '@deriv-com/analytics';
+
+import useIsGrowthbookIsLoaded from './useIsGrowthbookLoaded';
 
 const useDtraderV2Flag = () => {
     const { isGBLoaded: is_growthbook_loaded, isGBAvailable: is_gb_available } = useIsGrowthbookIsLoaded();
     const load_dtrader_module = is_growthbook_loaded || !is_gb_available;
 
-    const is_dtrader_v2 = isDTraderV2();
-    const { isMobile: is_mobile } = useDevice();
-    const is_feature_flag_active = Boolean(Analytics?.getFeatureValue('dtrader_v2_enabled', false));
-    const is_trade_or_contract_path =
-        location.pathname.startsWith(routes.trade) || location.pathname.startsWith('/contract/');
+    const { isMobile: is_mobile, isDesktop: is_desktop } = useDevice();
+    const is_dtrader_v2_mobile = isDtraderV2MobileEnabled(is_mobile);
+    const is_dtrader_v2_desktop = isDtraderV2DesktopEnabled(is_desktop);
+
+    const [dtrader_v2_enabled_mobile, setDtraderV2EnabledMobile] = useState(false);
+    const [dtrader_v2_enabled_desktop, setDtraderV2EnabledDesktop] = useState(false);
 
-    const [dtrader_v2_enabled, setDTraderV2Enabled] = useState(false);
     useEffect(() => {
-        if (is_growthbook_loaded || isDTraderV2()) {
-            setDTraderV2Enabled((is_dtrader_v2 || is_feature_flag_active) && is_mobile && is_trade_or_contract_path);
+        if (is_growthbook_loaded || is_dtrader_v2_mobile) {
+            setDtraderV2EnabledMobile(is_dtrader_v2_mobile);
+        }
+        if (is_growthbook_loaded || is_dtrader_v2_desktop) {
+            setDtraderV2EnabledDesktop(is_dtrader_v2_desktop);
         }
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [is_mobile, is_growthbook_loaded]);
```

Comment:
In the array with dep-s you have only is_mobile. Should is_desktop also bee there?